src/api/flow.ts

```diff
@@ -4,35 +4,9 @@ let generatorId = 0
 
 export type CancellablePromise<T> = Promise<T> & { cancel(): void }
 
-export interface FlowYield {
-    // fake, only for typing
-    "!!flowYield": undefined
-}
-
-export interface FlowReturn<T> {
-    // fake, only for typing
-    "!!flowReturn": T
-}
-
-// we skip promises that are the result of yielding promises (except if they use flowReturn)
-export type FlowReturnType<R> = IfAllAreFlowYieldThenVoid<
-    R extends FlowReturn<infer FR>
-        ? FR extends Promise<infer FRP>
-            ? FRP
-            : FR
-        : R extends Promise<any>
-        ? FlowYield
-        : R
->
-
-// we extract yielded promises from the return type
-export type IfAllAreFlowYieldThenVoid<R> = Exclude<R, FlowYield> extends never
-    ? void
-    : Exclude<R, FlowYield>
-
 export function flow<R, Args extends any[]>(
-    generator: (...args: Args) => IterableIterator<R>
-): (...args: Args) => CancellablePromise<FlowReturnType<R>> {
+    generator: (...args: Args) => Generator<any, R, any>
```

Comment:
In the mobx5 branch a union of Generator and Async generator was returned. What is the reason (or impact) of the difference?

Reply:
Support for async generators, forgot to bsckport that one 😨

Reply:
Should be fixed now